src/Microsoft.AspNetCore.Mvc.Razor/RazorPageActivator.cs

```diff
@@ -48,13 +33,15 @@ public class RazorPageActivator : IRazorPageActivator
             HtmlEncoder htmlEncoder,
             IModelExpressionProvider modelExpressionProvider)
         {
-            _activationInfo = new ConcurrentDictionary<Type, PageActivationInfo>();
-            _metadataProvider = metadataProvider;
-            _urlHelperAccessor = context => urlHelperFactory.GetUrlHelper(context);
-            _jsonHelperAccessor = context => jsonHelper;
-            _diagnosticSourceAccessor = context => diagnosticSource;
-            _htmlEncoderAccessor = context => htmlEncoder;
-            _modelExpressionProviderAccessor = context => modelExpressionProvider;
+            _razorPagePropertyActivator = new RazorPagePropertyActivator(
```

Comment:
The type is public, I'm not sure if we are ok making a breaking change modifying the ctor. At the same time, there's plenty of optimizations here that I'd rather not copy to RazorPages